scalableminds/webknossos
# Make ad-hoc meshes compatible with ND segmentations

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

@dieknolle3333 I think backend should be completed, though I have not tested it yet.

### Steps to test:
- open an annotation for a ND dataset
- create some segments
- load meshes ad hoc
- make sure all actions for single meshes, groups and multi-selects are working
- change nd-parameter(s) and make sure it's still working.
- when moving around the nth dimension, all loaded meshes should be shown.
- when removing a mesh in one nd-position, it should be gone everywhere.

### TODOs:
- [x] show only meshes of current `additionalCoordinates`
- [x] fix color changing
- [x] fix remove meshes
- [x] fix update meshes
- [x] fix context menu for single mesh in segment view
- [x] fix visibility toggle
- [x] fix segment multiselect
- [x] fix segment context menu in viewport
- [x] implement saga with while-true-loop that saves the previous additional coordinates
- [x] make sure segment view adjusts mesh actions
- [x] move `getAdditionalCoordinatesString` to method, e.g. in flycam_accessor
- [x] find a nicer way to access store in mesh saga
- [x] make sure its working for meshes without additional params
- [x] fix yaml.lock
- [x] fix case where segment in not present in current additional coordianate. see case where I brush new segment and then erase all pixels.
- [x] fix download in group menus
- [x] investiagte whether getMeshesForAdditionalCoordinates can be used in other places
- [x] fix that some meshes arent shown, e.g. after changing coordinates
- [x] only remove mesh in current add coord 

### Issues:
- fixes #7275 

------
(Please delete unneeded items, merge only when none are left open)
